src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -85,27 +85,32 @@ public final boolean init(Literal<?>[] args, int matchedPattern, ParseResult par
 			throw new IllegalStateException();
 		skriptEventInfo = (SkriptEventInfo<?>) syntaxElementInfo;
 
+		assert entryContainer != null; // cannot be null for non-simple structures
+		this.source = entryContainer.getSource();
+
+		// use default value for now
+		listeningBehavior = eventData.getListenerBehavior();
+
+		// initialize implementation
+		if (!init(args, matchedPattern, parseResult))
+			return false;
+
 		// evaluate whether this event supports listening to cancelled events
-		supportsListeningBehavior = false;
 		for (Class<? extends Event> eventClass : getEventClasses()) {
```

Comment:
if supportsListeningBehavior() is overridden to return true, we could skip this loop

Reply:
i was wondering if it would be good to still store the "default" value

Reply:
I suppose it doesn't hurt much to do it anyway